spec/requests/admin/users_controller_spec.rb

```diff
@@ -4,6 +4,810 @@
   let(:admin) { Fabricate(:admin) }
   let(:user) { Fabricate(:user) }
 
+  it 'is a subclass of AdminController' do
+    expect(Admin::UsersController < Admin::AdminController).to eq(true)
+  end
+
+  before do
+    sign_in(admin)
+  end
+
+  describe '#index' do
+    it 'returns success with JSON' do
+      get "/admin/users/list.json"
+      expect(response.status).to eq(200)
+      expect(JSON.parse(response.body)).to be_present
+    end
+
+    context 'when showing emails' do
+      it "returns email for all the users" do
+        get "/admin/users/list.json", params: { show_emails: "true" }
+        data = ::JSON.parse(response.body)
+        data.each do |user|
+          expect(user["email"]).to be_present
+        end
+      end
+
+      it "logs only 1 enty" do
+        expect(UserHistory.where(action: UserHistory.actions[:check_email], acting_user_id: admin.id).count).to eq(0)
+
+        get "/admin/users/list.json", params: { show_emails: "true" }
+
+        expect(UserHistory.where(action: UserHistory.actions[:check_email], acting_user_id: admin.id).count).to eq(1)
```

Comment:
There is no need to duplicate this line twice. We can do `expect {}.to change {}.by(...)`